runtime/src/accounts_db.rs

```diff
@@ -3507,13 +3533,26 @@ impl AccountsDb {
             }
         }
 
+        // Mark down these slots are about to be purged so that new attempts to scan these
+        // banks fail, and any ongoing scans over these slots will detect that they should abort
+        // their results
+        {
+            let mut locked_removed_slot_ids = self.accounts_index.removed_slot_ids.lock().unwrap();
```

Comment:
When we remove the state for a particular bank for slot `S`, we mark down the unique bank id of this particular `Bank` so that any ongoing scan over this bank will know their state may have been corrupted